rq/worker.py

```diff
@@ -861,25 +878,20 @@ def setup_work_horse_signals(self):
         signal.signal(signal.SIGINT, signal.SIG_IGN)
         signal.signal(signal.SIGTERM, signal.SIG_DFL)
 
-    def prepare_job_execution(self, job, heartbeat_ttl=None):
+    def prepare_job_execution(self, job):
         """Performs misc bookkeeping like updating states prior to
         job execution.
         """
-        if job.timeout == -1:
-            timeout = -1
-        else:
-            timeout = job.timeout or 180
-
-        if heartbeat_ttl is None:
-            heartbeat_ttl = self.job_monitoring_interval + 60
 
         with self.connection.pipeline() as pipeline:
             self.set_state(WorkerStatus.BUSY, pipeline=pipeline)
             self.set_current_job_id(job.id, pipeline=pipeline)
+            self.set_current_job_working_time(0, pipeline=pipeline)
```

Comment:
I think this needs to be moved to `monitor_work_horse()` after job has finished so this will be set to zero after job execution finishes.

Reply:
Good point. I added it to `monitor_work_horse`, but I also kept it here in case a work horse dies (due to an `OSError` for example) without resetting it. WDYT?